test/test_schema.py

```diff
@@ -290,3 +291,21 @@ def fn(q: InputC) -> None:
     }
     assert input_schema == expected_input_schema
     assert is_validate_dict(input_schema)
+
+
+def test_schema_to_base_model():
+    def examined_fn(first: list[dict]) -> dict:
+        pass
+
+    class ExpectedModel(BaseModel):
+        first: list[dict]
+
+    example_data = [{"x": 1, "y": 2}, {"a": 1, "b": 2}]
+
+    tested_model = js.schema_to_base_model(get_input_schema(examined_fn))
```

Comment:
But the signature of the input isn't a `BaseModel`? So this method would never be called on the input of the function you've defined above. This would work if the input to your function was:
```python
    def examined_fn(input: ExpectedModel) -> dict:
        pass
```

Reply:
To clarify: the `examined_fn` represents user-defined function passed to the `etl_uvicorn` to be wrapped. This test case is an isolated fragment of wrapping logic. 

As I understand logic from line 305 is to generate BaseModel from the signature in order to build the FastAPI argument. Here I would expect that BaseModel generated from this logic is equivalent and/or equal to the EpectedModel.

Is user-defined function required to have arguments defined as BaseModel? The provided examples show that function can contain regular typing annotations.

Reply:
The`schema_to_base_model` only gets called when the typing of the variable is detected to be a `BaseModel`. Such as in the example I posted above
```python
def examined_fn(input: ExpectedModel) -> dict: pass
   pass
```
But the way you've written it, it's trying to wrap the parameter `first: list[dict]` which does not conform to a `BaseModel` so this should break as intended.

Reply:
This test was created based on analysis of example you have provided. I have added more logs to improve visibility of each processing step (container and its classes name are redacted):
```
samplePlugin      | get_input_schema, sig: '(elements_dict: list[dict]) -> SamplePlugin.SamplePluginResponse', parameters: '[<Parameter "elements_dict: list[dict]">]'
samplePlugin      | parameters_to_json_schema: parameter 'elements_dict: list[dict]', schema: '{'type': 'array', 'items': {'type': 'object'}}'
samplePlugin      | parameters_to_json_schema: return '{'type': 'object', 'required': ['elements_dict'], 'properties': {'elements_dict': {'type': 'array', 'items': {'type': 'object'}}}}'
samplePlugin      | schema_to_base_model schema: '{'type': 'object', 'required': ['elements_dict'], 'properties': {'elements_dict': {'type': 'array', 'items': {'type': 'object'}}}}'
samplePlugin      | schema_to_base_model schema item: 'type', 'object'
samplePlugin      | schema_to_base_model schema item: 'required', '['elements_dict']' 
samplePlugin      | schema_to_base_model schema item: 'properties', '{'elements_dict': {'type': 'array', 'items': {'type': 'object'}}}'
samplePlugin      | schema_to_base_model Create model: 'reconstructed_model', inputs: '{}'
```

As you can see the input of the provided function in the example is `list[dict]`. Code path is following:
1. `func` is defined here: [api_generator.py#L35](https://github.com/Unstructured-IO/unstructured-platform-plugins/blob/jakub/debug-invoke-inputs/unstructured_platform_plugins/etl_uvicorn/api_generator.py#L35) by call to the `get_func`.
2. `get_func` defined in [utils.py#L13](https://github.com/Unstructured-IO/unstructured-platform-plugins/blob/jakub/debug-invoke-inputs/unstructured_platform_plugins/etl_uvicorn/utils.py#L13) does not do any checks regarding input being of type `BaseModel`.
3. Then the `func` is passed directly to the code which builds the `BaseModel` from its signature: [api_generator.py#L49](https://github.com/Unstructured-IO/unstructured-platform-plugins/blob/jakub/debug-invoke-inputs/unstructured_platform_plugins/etl_uvicorn/api_generator.py#L49)

Where is the requirement for input of `func` defined to be `BaseModel`?

I would like to ask for the following:
1. Please provide working example with instructions how to run it and attach logs from sample run.
2. Please define clear interface when there is some strict requirement.